provider/core/src/lib.rs

```diff
@@ -170,10 +170,11 @@ pub mod prelude {
     pub use crate::request::*;
     #[doc(no_inline)]
     pub use crate::{
-        BoundDataProvider, DataError, DataErrorKind, DataLocale, DataMarker, DataMarkerAttributes,
-        DataMarkerInfo, DataPayload, DataProvider, DataRequest, DataRequestMetadata, DataResponse,
-        DataResponseMetadata, DryDataProvider, DynamicDataMarker, DynamicDataProvider,
-        DynamicDryDataProvider, IterableDataProvider, IterableDynamicDataProvider, ResultDataError,
+        marker::locale_for_data_marker, BoundDataProvider, DataError, DataErrorKind, DataLocale,
```

Comment:
not a big fan of having free functions floating around in the prelude. rust style is to always qualify functions.

Reply:
Me either, but I don't really see an alternative that doesn't make everyone need to fully qualify. This is needed often enough.

We can turn it into a defaulted method on DataMarker if we want but it seems like we're trying to move away from that model.

Reply:
Let's move both make_locale _and_ bind into a DataMarkerUtils trait